src/expr/src/linear.rs

```diff
@@ -1009,6 +1009,7 @@ pub fn memoize_expr(
                 MirScalarExpr::Column(_) | MirScalarExpr::Literal(_, _) => {
                     // Literals do not need to be memoized.
                 }
+                MirScalarExpr::CallNullary(_func @ NullaryFunc::MzLogicalTimestamp) => {}
```

Comment:
should I just call `contains_temporal()` at the beginning of this function?

Reply:
I think that is probably the "better fix". If we had e.g. `mz_logical_timestamp() + 5` we wouldn't want to memoize that. Though, at the same time we'd be happy to allow memoization of the branches that don't contain `mz_logical_timestamp()` (e.g. if that `5` was a more interesting expression). The `If` logic does something similar, where it will descend down expressions not memoizing, but still looking for opportunities to re-use already-memoized expressions.

I think keeping things conservative and simple is a great start (i.e. guard the whole thing by `contains_temporal()`), but if you want to flesh out further CSE opportunities then great! But I think just the above could potentially still be erroneous if `mz_logical_timestamp()` participates in a more complex re-used expression.